src/blockdata/script.rs

```diff
@@ -433,42 +439,42 @@ impl Script {
     }
 
     /// Write the assembly decoding of the script to the formatter.
-    pub fn fmt_asm(&self, f: &mut dyn fmt::Write) -> fmt::Result {
+    pub fn fmt_asm(script: &[u8], f: &mut dyn fmt::Write) -> fmt::Result {
```

Comment:
Why not `impl AsRef<[u8]>`?

Reply:
Can we already do `impl` with 1.29?  I wouldn't be too opposed to making things more generic like that, but otoh if something already is `AsRef<[u8]>` you can just call `as_ref()` on it when calling `fmt_asm`, it's just some very marginal convenience.

It's sad the following doesn't work:

```rust
struct Script(Vec<u8>);

impl AsRef<[u8]> for Script {
    fn as_ref(&self) -> &[u8] {
        &self.0
    }
}

impl Script {
    fn asm(self: impl AsRef<[u8]>) -> String {
        unimplemented!();
    }
}
```

This unfortunately only seems to work for types that directly deref to `Self`:
```
   = note: type of `self` must be `Self` or a type that dereferences to it
   = help: consider changing to `self`, `&self`, `&mut self`, `self: Box<Self>`, `self: Rc<Self>`, `self: Arc<Self>`, or `self: Pin<P>` (where P is one of the previous types except `Self`)
```

Reply:
Yeah I agree it it's a marginal benefit for a barely used API. Better keep it more readable.